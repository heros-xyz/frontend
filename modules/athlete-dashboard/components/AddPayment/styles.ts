export const styles = {
  textarea: {
    paddingLeft: 0,
    borderRadius: 0,
    borderColor: "#ADADAD",
  },
};
