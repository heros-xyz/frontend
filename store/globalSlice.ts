import { PayloadAction, createSlice } from "@reduxjs/toolkit";
import { HYDRATE } from "next-redux-wrapper";

const initialState = {
  isLoading: false,
  isMute: true,
  isPlaying: false,
  accessToken: '',
  refreshToken: ''
};

export const appState = createSlice({
  name: "appState",
  initialState: initialState,
  reducers: {
    startLoading: (state) => {
      state.isLoading = true;
    },
    finishLoading: (state) => {
      state.isLoading = false;
    },
    playVideo: (state) => {
      state.isPlaying = true;
    },
    pauseVideo: (state) => {
      state.isPlaying = false;
    },
    setAccessToken: (state, action: PayloadAction<string>) => {
      state.accessToken = action.payload
    },
    setRefreshToken: (state, action: PayloadAction<string>) => {
      state.refreshToken = action.payload
    }
  },
  extraReducers: {
    [HYDRATE]: (state, action) => {
      if (action.payload.appState.accessToken) {
        state.accessToken = action.payload.appState.accessToken;
      }

      if (action.payload.appState.refreshToken) {
        state.refreshToken = action.payload.appState.refreshToken;
      }
    }
  }
});

export const { startLoading, finishLoading, setAccessToken, setRefreshToken } = appState.actions;

export default appState.reducer;
