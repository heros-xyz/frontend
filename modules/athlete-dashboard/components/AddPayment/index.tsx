import {
  Box,
  Button,
  Center,
  Flex,
  Input,
  Text,
  Textarea,
} from "@chakra-ui/react";
import { useFormik } from "formik";
import { useState } from "react";
import { ArrowLeft } from "@/components/svg/ArrowLeft";
import DateSelect from "@/components/ui/DateSelect";
import ErrorMessage from "@/components/common/ErrorMessage";
import SelectGender from "@/components/ui/SelectGender";
import Select from "@/components/common/Select";
import { useGetNationalityQuery } from "@/api/global";
import { filterSelectOptions } from "@/utils/functions";
import { styles } from "./styles";
import {
  initialAddpayment,
  initialValues,
  validationSchema,
} from "../../constants";

interface IProp {
  onSubmit: (values: initialAddpayment) => void;
  initialValues?: initialAddpayment;
}

const AddPayment: React.FC<IProp> = ({ onSubmit }) => {
  const [nation, setNation] = useState<{ value: string; label: string }>({
    value: "",
    label: "",
  });
  const formik = useFormik({
    initialValues: initialValues,
    validationSchema: validationSchema,
    onSubmit: (values) => {
      onSubmit(values);
    },
  });

  const { data: nationalityList } = useGetNationalityQuery("");
  return (
    <Center
      color="white"
      flexDirection="column"
      p={{ base: "5", xl: "19rem" }}
      pt={{ xl: "3.75rem" }}
    >
      <Box w={{ xl: "30rem" }}>
        <Box w="full" fontWeight="bold">
          <ArrowLeft
            verticalAlign=""
            w={{ base: "14px", xl: "18px" }}
            h={{ base: "14px", xl: "18px" }}
          />
          <Text as="span" ml="6" fontSize={{ base: "xl", xl: "2xl" }}>
            Add Payment Method
          </Text>
        </Box>
        <Box w="full" fontSize={{ base: "sm", xl: "md" }}>
          <form onSubmit={formik.handleSubmit}>
            <Box mt={{ base: "5", xl: "8" }}>
              <Box fontWeight="medium">
                Legal First Name
                <Text as="span" color="error.dark">
                  {" "}
                  *
                </Text>
              </Box>
              <Input
                variant="flushed"
                w="full"
                id="firstName"
                name="firstName"
                onChange={formik.handleChange}
                value={formik.values.firstName}
                isInvalid={Boolean(
                  formik.errors.firstName && formik.touched.firstName
                )}
              />
              <ErrorMessage
                fontSize={{ xl: "sm" }}
                mt={0.5}
                condition={formik.errors.firstName && formik.touched.firstName}
                errorMessage={formik.errors.firstName}
              />
            </Box>
            <Box mt={{ base: "5", xl: "8" }}>
              <Box fontWeight="medium">
                Legal Last Name
                <Text as="span" color="error.dark">
                  {" "}
                  *
                </Text>
              </Box>
              <Input
                variant="flushed"
                w="full"
                id="lastName"
                name="lastName"
                onChange={formik.handleChange}
                value={formik.values.lastName}
                isInvalid={Boolean(
                  formik.errors.lastName && formik.touched.lastName
                )}
              />
              <ErrorMessage
                mt={0.5}
                condition={formik.errors.lastName && formik.touched.lastName}
                errorMessage={formik.errors.lastName}
              />
            </Box>
            <Box mt={{ base: "5", xl: "8" }}>
              <Box fontWeight="medium">
                Date of Birth
                <Text as="span" color="error.dark">
                  {" "}
                  *
                </Text>
              </Box>
              <Flex
                justifyContent="space-between"
                gap={{ base: 3, lg: 8 }}
                mt="2"
              >
                <DateSelect
                  date={formik.values?.dateOfBirth}
                  onChange={(value) =>
                    formik.setFieldValue("dateOfBirth", value)
                  }
                  submitted={!!formik.submitCount}
                  zIndex={20}
                />
              </Flex>
              <ErrorMessage
                mt={0.5}
                condition={Boolean(
                  formik.submitCount && formik.errors?.dateOfBirth
                )}
                errorMessage={formik.errors?.dateOfBirth}
              />
            </Box>
            <Box mt={{ base: "5", xl: "8" }}>
              <Box fontWeight="medium" mb="3">
                Gender
                <Text as="span" color="error.dark">
                  {" "}
                  *
                </Text>
              </Box>
              <SelectGender
                value={formik.values.gender}
                onChange={(value) => formik.setFieldValue("gender", value)}
                errorMessage={
                  formik.errors.gender && formik.touched.gender
                    ? "This is a required field"
                    : ""
                }
                flexRow={true}
                bgColor="primary"
              />
            </Box>
            <Box mt={{ base: "5", xl: "8" }}>
              <Box fontWeight="medium">
                Nationality
                <Text as="span" color="error.dark">
                  {" "}
                  *
                </Text>
              </Box>
              <Box mt={{ base: "2", xl: "4" }}>
                <Select
                  value={nation}
                  options={nationalityList}
                  filterSelectOptions={filterSelectOptions}
                  onChange={(el: { value: string; label: string }) => {
                    formik.setFieldValue("nationality", el.value);
                    setNation(el);
                  }}
                  errorMessage={"This is a required field"}
                  isInvalid={Boolean(
                    formik.errors.nationality && formik.touched.nationality
                  )}
                />
              </Box>
            </Box>
            <Box mt={{ base: "5", xl: "8" }}>
              <Box fontWeight="medium">
                My Story
                <Text as="span" color="error.dark">
                  {" "}
                  *
                </Text>
              </Box>
              <Text
                color="secondary"
                fontSize={{ base: "xs", xl: "md" }}
                mt="1"
              >
                This is the first thing potential patrons will see when they
                land on your page, so make sure you paint a compelling picture
                of how they can join you on this journey.
              </Text>
              <Textarea
                id="myStory"
                name="myStory"
                variant="flushed"
                onChange={formik.handleChange}
                value={formik.values.myStory}
                placeholder="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."
                resize={"none"}
                size={{ base: "sm", xl: "md" }}
                minH={20}
                style={styles.textarea}
                isInvalid={Boolean(
                  formik.errors.myStory && formik.touched.myStory
                )}
                mt="2"
              />
              <ErrorMessage
                mb={8}
                condition={formik.errors.myStory && formik.touched.myStory}
                errorMessage={formik.errors.myStory}
              />
            </Box>
            <Button
              bg="secondary"
              color="primary"
              w={{ base: "100%", xl: "auto" }}
              mt={"12"}
              fontWeight={"bold"}
              type="submit"
              fontSize={{ xl: "xl" }}
              float={{ xl: "right" }}
            >
              SAVE
            </Button>
          </form>
        </Box>
      </Box>
    </Center>
  );
};

export default AddPayment;
