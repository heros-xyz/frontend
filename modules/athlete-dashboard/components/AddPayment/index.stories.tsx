import { Box } from "@chakra-ui/react";
import type { Meta, StoryObj } from "@storybook/react";
import { Provider } from "react-redux";
import { store } from "@/store";
import AddPayment from ".";

const meta: Meta<typeof AddPayment> = {
  title: "Athlete/AddPayment",
  component: AddPayment,
  tags: ["autodocs"],
  parameters: {
    layout: "fullscreen",
  },
};

export default meta;
type Story = StoryObj<typeof AddPayment>;

export const AddPaymentComponent: Story = {
  render: (args) => (
    <Provider store={store}>
      <Box minH="100vh" bg="primary">
        <AddPayment {...args} />
      </Box>
    </Provider>
  ),
};
