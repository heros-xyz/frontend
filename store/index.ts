import { configureStore } from "@reduxjs/toolkit";
import { createWrapper } from "next-redux-wrapper";
import { TypedUseSelectorHook, useDispatch, useSelector } from "react-redux";
import { setupListeners } from "@reduxjs/toolkit/query";
import globalSlice from "./globalSlice";

export const store = configureStore({
  reducer: {
    appState: globalSlice,

  },
  devTools: true,
  middleware: (gDM) => {
    return gDM().concat(

    );
  },
});

const makeStore = () => store;

export type AppStore = ReturnType<typeof makeStore>;
export type AppState = ReturnType<AppStore["getState"]>;
export type AppDispatch = AppStore["dispatch"];

export const wrapper = createWrapper<AppStore>(makeStore);
export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<AppState> = useSelector;

setupListeners(store.dispatch);
