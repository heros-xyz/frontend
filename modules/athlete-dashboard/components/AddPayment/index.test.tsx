import { Box } from "@chakra-ui/react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "@/store";
import AddPayment from ".";

it("renders homepage unchanged", () => {
  const onChangeStep = (step: number) => {};
  const { container } = render(
    <Provider store={store}>
      <Box minH="100vh" bg="primary">
        <AddPayment onSubmit={() => onChangeStep(2)} />
      </Box>
    </Provider>
  );
  expect(container).toMatchSnapshot();
});
