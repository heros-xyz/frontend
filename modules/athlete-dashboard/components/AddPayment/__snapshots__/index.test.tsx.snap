// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <div
    class="css-1gxxz4r"
  >
    <div
      class="css-tvi62e"
    >
      <div
        class="css-1bkgb7h"
      >
        <div
          class="css-soukc5"
        >
          <svg
            class="chakra-icon css-ltjg40"
            focusable="false"
            viewBox="0 0 32 32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M26 16L6 16M6 16L15 7M6 16L15 25"
              stroke="#1E16C1"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
            />
          </svg>
          <span
            class="chakra-text css-141yne6"
          >
            Add Payment Method
          </span>
        </div>
        <div
          class="css-1ex591z"
        >
          <form>
            <div
              class="css-1dd5wzm"
            >
              <div
                class="css-1xnzs7r"
              >
                Legal First Name
                <span
                  class="chakra-text css-1b8yx6g"
                >
                   
                  *
                </span>
              </div>
              <input
                class="chakra-input css-13rbvqk"
                id="firstName"
                name="firstName"
                value=""
              />
            </div>
            <div
              class="css-1dd5wzm"
            >
              <div
                class="css-1xnzs7r"
              >
                Legal Last Name
                <span
                  class="chakra-text css-1b8yx6g"
                >
                   
                  *
                </span>
              </div>
              <input
                class="chakra-input css-13rbvqk"
                id="lastName"
                name="lastName"
                value=""
              />
            </div>
            <div
              class="css-1dd5wzm"
            >
              <div
                class="css-1xnzs7r"
              >
                Date of Birth
                <span
                  class="chakra-text css-1b8yx6g"
                >
                   
                  *
                </span>
              </div>
              <div
                class="css-1ulhi5g"
              >
                <div
                  class="css-17eahu8"
                >
                  <div
                    class=" css-n9lnso-SelectContainer"
                  >
                    <span
                      class="css-1f43avz-a11yText-A11yText"
                      id="react-select-:r0:-live-region"
                    />
                    <span
                      aria-atomic="false"
                      aria-live="polite"
                      aria-relevant="additions text"
                      class="css-1f43avz-a11yText-A11yText"
                    />
                    <div
                      class=" css-6o1t9b-control"
                    >
                      <div
                        class=" css-15ki4p9-ValueContainer"
                      >
                        <div
                          class="css-xxdqwu"
                        >
                          <div
                            class="css-138yobv"
                          >
                            DD
                          </div>
                          <div
                            class=" css-1fs6gt4-Input"
                            data-value=""
                          >
                            <input
                              aria-autocomplete="list"
                              aria-describedby="react-select-:r0:-placeholder"
                              aria-expanded="false"
                              aria-haspopup="true"
                              autocapitalize="none"
                              autocomplete="off"
                              autocorrect="off"
                              class=""
                              id="react-select-:r0:-input"
                              role="combobox"
                              spellcheck="false"
                              style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                              tabindex="0"
                              type="text"
                              value=""
                            />
                          </div>
                        </div>
                      </div>
                      <div
                        class=" css-1hb7zxy-IndicatorsContainer"
                      >
                        <span
                          class=" css-1xjgjl1-IndicatorSeparator"
                        />
                        <svg
                          class="chakra-icon css-h50qoc"
                          focusable="false"
                          viewBox="0 0 20 20"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M16.25 6.875L10 13.125L3.75 6.875"
                            stroke="currentColor"
                            strokeLinecap="round"
                            strokeLinejoin="round"
                            strokeWidth="1.5"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="css-17eahu8"
                >
                  <div
                    class=" css-n9lnso-SelectContainer"
                  >
                    <span
                      class="css-1f43avz-a11yText-A11yText"
                      id="react-select-:r1:-live-region"
                    />
                    <span
                      aria-atomic="false"
                      aria-live="polite"
                      aria-relevant="additions text"
                      class="css-1f43avz-a11yText-A11yText"
                    />
                    <div
                      class=" css-6o1t9b-control"
                    >
                      <div
                        class=" css-15ki4p9-ValueContainer"
                      >
                        <div
                          class="css-xxdqwu"
                        >
                          <div
                            class="css-138yobv"
                          >
                            MM
                          </div>
                          <div
                            class=" css-1fs6gt4-Input"
                            data-value=""
                          >
                            <input
                              aria-autocomplete="list"
                              aria-describedby="react-select-:r1:-placeholder"
                              aria-expanded="false"
                              aria-haspopup="true"
                              autocapitalize="none"
                              autocomplete="off"
                              autocorrect="off"
                              class=""
                              id="react-select-:r1:-input"
                              role="combobox"
                              spellcheck="false"
                              style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                              tabindex="0"
                              type="text"
                              value=""
                            />
                          </div>
                        </div>
                      </div>
                      <div
                        class=" css-1hb7zxy-IndicatorsContainer"
                      >
                        <span
                          class=" css-1xjgjl1-IndicatorSeparator"
                        />
                        <svg
                          class="chakra-icon css-h50qoc"
                          focusable="false"
                          viewBox="0 0 20 20"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M16.25 6.875L10 13.125L3.75 6.875"
                            stroke="currentColor"
                            strokeLinecap="round"
                            strokeLinejoin="round"
                            strokeWidth="1.5"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="css-17eahu8"
                >
                  <div
                    class=" css-n9lnso-SelectContainer"
                  >
                    <span
                      class="css-1f43avz-a11yText-A11yText"
                      id="react-select-:r2:-live-region"
                    />
                    <span
                      aria-atomic="false"
                      aria-live="polite"
                      aria-relevant="additions text"
                      class="css-1f43avz-a11yText-A11yText"
                    />
                    <div
                      class=" css-6o1t9b-control"
                    >
                      <div
                        class=" css-15ki4p9-ValueContainer"
                      >
                        <div
                          class="css-xxdqwu"
                        >
                          <div
                            class="css-138yobv"
                          >
                            YYYY
                          </div>
                          <div
                            class=" css-1fs6gt4-Input"
                            data-value=""
                          >
                            <input
                              aria-autocomplete="list"
                              aria-describedby="react-select-:r2:-placeholder"
                              aria-expanded="false"
                              aria-haspopup="true"
                              autocapitalize="none"
                              autocomplete="off"
                              autocorrect="off"
                              class=""
                              id="react-select-:r2:-input"
                              role="combobox"
                              spellcheck="false"
                              style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                              tabindex="0"
                              type="text"
                              value=""
                            />
                          </div>
                        </div>
                      </div>
                      <div
                        class=" css-1hb7zxy-IndicatorsContainer"
                      >
                        <span
                          class=" css-1xjgjl1-IndicatorSeparator"
                        />
                        <svg
                          class="chakra-icon css-h50qoc"
                          focusable="false"
                          viewBox="0 0 20 20"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M16.25 6.875L10 13.125L3.75 6.875"
                            stroke="currentColor"
                            strokeLinecap="round"
                            strokeLinejoin="round"
                            strokeWidth="1.5"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="css-1dd5wzm"
            >
              <div
                class="css-qhhr1q"
              >
                Gender
                <span
                  class="chakra-text css-1b8yx6g"
                >
                   
                  *
                </span>
              </div>
              <div
                class="css-1a3i73t"
              >
                <div
                  class="css-15s5pib"
                >
                  <p
                    class="chakra-text css-1a4koi3"
                  >
                    Male
                  </p>
                </div>
                <div
                  class="css-15s5pib"
                >
                  <p
                    class="chakra-text css-1a4koi3"
                  >
                    Female
                  </p>
                </div>
                <div
                  class="css-15s5pib"
                >
                  <p
                    class="chakra-text css-1a4koi3"
                  >
                    Other
                  </p>
                </div>
              </div>
            </div>
            <div
              class="css-1dd5wzm"
            >
              <div
                class="css-1xnzs7r"
              >
                Nationality
                <span
                  class="chakra-text css-1b8yx6g"
                >
                   
                  *
                </span>
              </div>
              <div
                class="css-ytg8w7"
              >
                <div
                  class="css-s2ct51"
                >
                  <div
                    class=" css-n9lnso-SelectContainer"
                  >
                    <span
                      class="css-1f43avz-a11yText-A11yText"
                      id="react-select-:r3:-live-region"
                    />
                    <span
                      aria-atomic="false"
                      aria-live="polite"
                      aria-relevant="additions text"
                      class="css-1f43avz-a11yText-A11yText"
                    />
                    <div
                      class=" css-6o1t9b-control"
                    >
                      <div
                        class=" css-15ki4p9-ValueContainer"
                      >
                        <div
                          class="css-xxdqwu"
                        >
                          <div
                            class="css-138yobv"
                          >
                            Select...
                          </div>
                          <div
                            class=" css-1fs6gt4-Input"
                            data-value=""
                          >
                            <input
                              aria-autocomplete="list"
                              aria-describedby="react-select-:r3:-placeholder"
                              aria-expanded="false"
                              aria-haspopup="true"
                              autocapitalize="none"
                              autocomplete="off"
                              autocorrect="off"
                              class=""
                              id="react-select-:r3:-input"
                              role="combobox"
                              spellcheck="false"
                              style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                              tabindex="0"
                              type="text"
                              value=""
                            />
                          </div>
                        </div>
                      </div>
                      <div
                        class=" css-1hb7zxy-IndicatorsContainer"
                      >
                        <span
                          class=" css-1xjgjl1-IndicatorSeparator"
                        />
                        <svg
                          class="chakra-icon css-h50qoc"
                          focusable="false"
                          viewBox="0 0 20 20"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M16.25 6.875L10 13.125L3.75 6.875"
                            stroke="currentColor"
                            strokeLinecap="round"
                            strokeLinejoin="round"
                            strokeWidth="1.5"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="css-1dd5wzm"
            >
              <div
                class="css-1xnzs7r"
              >
                My Story
                <span
                  class="chakra-text css-1b8yx6g"
                >
                   
                  *
                </span>
              </div>
              <p
                class="chakra-text css-1f5yt9g"
              >
                This is the first thing potential patrons will see when they land on your page, so make sure you paint a compelling picture of how they can join you on this journey.
              </p>
              <textarea
                class="chakra-textarea css-1puq9p5"
                id="myStory"
                name="myStory"
                placeholder="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."
                style="padding-left: 0px; border-radius: 0; border-color: #adadad;"
              />
            </div>
            <button
              class="chakra-button css-1ydnnxh"
              type="submit"
            >
              SAVE
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
